### ISSUE TITLE
VRP with a single location

### ISSUE THREAD

**Issue by senhalil** [VROOM-Project/vroom#291]
The following VRP problem with a single location (note that all location indices are 0) but with a 2x2 matrix returns the expected solution with zero cost. 

However, if the matrix is replaced by a 1x1 matrix, then vroom tries to connect to OSRM and fail with the following error `{"code":3,"error":"Failed to connect to 0.0.0.0:5000"}` (since there is no lat/lon, even if the connection succeeds, the query would fail anyway). 

I believe the offender is  `if (_matrix.size() < 2) {` check on src/structures/vroom/input/input.cpp:332 inside solve function. 

Is the check against `2` is necessary? It seems to me that `1` does the job well. 

```
# input
{
  "vehicles": [
    {
      "id": 0,
      "start_index": 0,
      "end_index": 0
    }
  ],
  "jobs": [
    {
      "id": 0,
      "location_index": 0
    }
  ],
  "matrix": [
    [
      0,
      6232
    ],
    [
      6239,
      0
    ]
  ]
}
```
```
# output
{
  "code": 0,
  "summary": {
    "cost": 0,
    "unassigned": 0,
    "service": 0,
    "duration": 0,
    "waiting_time": 0,
    "computing_times": {
      "loading": 0,
      "solving": 0
    }
  },
  "unassigned": [],
  "routes": [
    {
      "vehicle": 0,
      "cost": 0,
      "service": 0,
      "duration": 0,
      "waiting_time": 0,
      "steps": [
        {
          "type": "start",
          "arrival": 0,
          "duration": 0
        },
        {
          "type": "job",
          "job": 0,
          "service": 0,
          "waiting_time": 0,
          "arrival": 0,
          "duration": 0
        },
        {
          "type": "end",
          "arrival": 0,
          "duration": 0
        }
      ]
    }
  ]
}
```

**Comment by senhalil** [comment]
I went ahead and opened a pull-request just in case.

**Comment by jcoupey** [comment]
Kind of an edge case, but definitely a bug. The `_matrix.size() < 2` check is a poor man's way to test if a custom matrix has been provided in input. It may have been OK back in the days (with TSP-only code) because we probably had other ways to enforce having several locations.

A clean fix would be to add a boolean member along the line of `_has_custom_matrix` and use it as a flag for the above test. I think it would be sufficient to have it set to `false` in the `Input` ctor and switch to `true` whenever `Input::set_matrix` is called.

**Comment by senhalil** [comment]
I see the fix you suggested. I will modify the pull request accordingly.

**Comment by jcoupey** [comment]
Hi @senhalil what's your schedule on this?

Just asking because I'd like to include this fix in the next release candidate, hopefully in the next few days (also #295 would benefit from having a flag member to know whether a custom matrix has been provided).

**Comment by senhalil** [comment]
Sorry for the radio silence. I'll look into it now.

**Comment by senhalil** [comment]
Ready for review.

**Comment by jcoupey** [comment]
Thanks!

**Comment by senhalil** [comment]
> This is fixed in #298. Thanks @senhalil for reporting and submitting a PR!

Happy to help!
